src/backend/joanie/payment/models.py

```diff
@@ -1 +1,465 @@
+        """
+        Update or create the invoice context for all languages.
+
+        In order to save space, we want to generate related document only on
+        demand. Furthermore, we need to store product description to prevent
+        inconsistency in case of product would be updated. That's why we store
+        a product information for all active language into the invoice, then we
+        are able to generate invoice document with consistent data only on
+        demand.
+
+        Saving is left to the caller.
+        """
+
+        context = {}
+        related_product = self.related_order.product
+        for language, __ in settings.LANGUAGES:
+            with switch_language(related_product, language):
+                context[language] = {
+                    "order": {
+                        "product": {
+                            "name": related_product.title,  # pylint: disable=no-member
+                            "description": related_product.description,  # noqa pylint: disable=no-member,line-too-long
+                        }
+                    }
+                }
+
+        self.localized_context = context
+
+    def get_document_context(self, language_code=None):
+        """
+        Build the invoice document context for the given language.
+        If no language_code is provided, we use the active language.
+        """
+        language_settings = get_language_settings(language_code or get_language())
+
+        vat = D(settings.JOANIE_VAT)
+        total_amount = self.total.amount  # pylint: disable=no-member
+        vat_amount = total_amount * vat / 100
+        net_amount = total_amount - vat_amount
+        currency = get_currency_symbol(
+            self.total.currency.code  # pylint: disable=no-member
+        )
+
+        base_context = {
+            "metadata": {
+                "issued_on": str(self.updated_on),
+                "reference": self.reference,
+                "type": self.type,
+            },
+            "order": {
+                "amount": {
+                    "currency": currency,
+                    "subtotal": str(net_amount),
+                    "total": str(total_amount),
+                    "vat_amount": str(vat_amount),
+                    "vat": str(vat),
+                },
+                "company": settings.JOANIE_INVOICE_COMPANY_CONTEXT,
+                "customer": {
+                    "address": self.recipient_address,
+                    "name": self.recipient_name,
+                },
+                "seller": {
+                    "address": settings.JOANIE_INVOICE_SELLER_ADDRESS,
+                },
+            },
+        }
+
+        try:
+            localized_context = self.localized_context[language_settings["code"]]
+        except KeyError:
+            # - Otherwise use the first language fallback
+            localized_context = self.localized_context[
+                language_settings["fallbacks"][0]
```

Comment:
how are you sure that this one exists?

Reply:
Until you explictly defined `fallbacks` property to `[]` or `None` for the current language, this value should exist (by default, django parler uses `LANGUAGE_CODE` if no default fallback has been defined).

Do you prefer that I fallback here to `settings.LANGUAGE_CODE` in case of `fallbacks[0]` raises an `IndexError`?

Reply:
I would say yes if you manage to make it happen in a test with override_settings?

Reply:
Finally, after some tests, I decided to use the same logic used by django parler with TranslatedField and `any_language=True`: I fallback to the first available localized context. But in fact, this is really an edge case because `KeyError` is raised only when all fallbacks has been voluntary sets to `None` or `[]`.

```python
        try:
            localized_context = self.localized_context[language_settings["code"]]
        except KeyError:
            # - Otherwise use the first entry of the context
            localized_context = list(self.localized_context.values())[0]
```

Reply:
:+1: